File: C#/Snake Game Clone in CS (2014)/SnakeGameCS/Walls.cs
/////////////////
//Walls class
class walls
{
	//Constructor
	public walls(char iR)
	{
		//Set representation
		r = iR;
	}

	//Draw
	public void draw(graphics target)
	{
		//Cycle y
		for(int y = 0; y < target.data.height; ++y)
		{
			//Cycle x
			for(int x = 0; x < target.data.width; ++x)
			{
				if((x == 0) || (x == target.data.width-1) || (y == 0) || (y == target.data.height-1))
					target.data[y, x] = r;
			}
		}
	}

	public char r; //Representation
};
/////////////////

File: C++ VS2013/OpenGL with Shaders (2014-2015)/Snake (GLSL) (2015)/Snake (GLSL)/Walls.cpp
#include "Walls.h"

//Constructor
Walls::Walls(unsigned int Width, unsigned int Height, Color C):
c(C)
{
	//Create horizontal walls
	for(unsigned int count = 0; count < Width; ++count)
	{
		walls.push_back(Point((float)count, 0));
		walls.push_back(Point((float)count, (float)Height));
	}
	
	//Create vertical walls
	for(unsigned int count = 0; count < Height; ++count)
	{
		walls.push_back(Point(0, (float)count));
		walls.push_back(Point((float)Width, (float)count));
	}
}

//Get
const std::vector<Point> &Walls::Get()
{
	return walls;
}

//Collision
bool Walls::Collision(Goal &Target)
{
	for(unsigned int count = 0; count < walls.size(); ++count)
	{
		if((walls.at(count).x == Target.x) && (walls.at(count).y == Target.y))
			return true;
	}
	return false;
}